aw-server/src/android/mod.rs

```diff
@@ -82,9 +82,10 @@ pub mod android {
         output.into_raw()
     }
 
-    unsafe fn jstring_to_string(env: &JNIEnv, string: JString) -> String {
-        let c_str = CStr::from_ptr(env.get_string(string).expect("invalid string").as_ptr());
-        String::from(c_str.to_str().unwrap())
+    unsafe fn jstring_to_string(env: &JNIEnv, string: JString) -> Option<String> {
+        let java_str = env.get_string(string).ok()?;
```

Comment:
BTW, did you find out how these can fail?

Reply:
I only figured out they were failing by using a lot of trace logs. Then updated them didn't investigate much about how exactly they were failing.